firestore/client.go

```diff
@@ -234,7 +279,9 @@ func (c *Client) getAll(ctx context.Context, docRefs []*DocumentRef, tid []byte,
 		req.ConsistencySelector = &pb.BatchGetDocumentsRequest_Transaction{Transaction: tid}
 	}
 
-	streamClient, err := c.c.BatchGetDocuments(withResourceHeader(ctx, req.Database), req)
+	batchGetDocsCtx := withResourceHeader(ctx, req.Database)
+	batchGetDocsCtx = withRequestParamsHeader(batchGetDocsCtx, reqParamsHeaderVal(c.path()))
```

Comment:
Relating to the comment above, If using the `gax-go` helper these two lines would be one:

```go
batchGetDocsCtx := gax.InsertMetadataIntoOutgoingContext(ctx, reqParamsHeader, requestParams, resourcePrefixHeader, req.Database)
```